### PR TITLE
enhancement/post-process-ingestion-models

### LINKED ISSUE #31: Create function to allow post processing of the produced CDP Ingestion Models
### Feature Description

_A clear and concise description of the feature you're requesting._

Add a function to the base `LegistarScraper` that allows processing and or extra validation of the produced CDP ingestion model data.

### Use Case

_Please provide a use case to help us understand your request in context._

Each person returned from legistar comes with a webpage URL but is missing their picture uri. In the post processing function a scraper author could take these URLs and scrape them to get the picture URL.

### Solution

_Please describe your ideal solution._

Have a function added to the `LegistarScraper` class called like:

```python
def post_process_ingestion_models(events: List[EventIngestionModel]) -> List[EventIngestionModel]:
    return events
```

The default is simply to return the events (don't process anything) but in inheriting scrapers, the author can add more to this function.

### Alternatives

_Please describe any alternatives you've considered, even if you've dismissed them._

### PR DESCRIPTION
**Pull request by dphoria**
### Link to Relevant Issue

This pull request resolves #31 

### Description of Changes

Added `post_process_ingestion_models(events: List[EventIngestionModel])` to `LegistarScraper`. Base implementation does nothing; returns input `events` as-is. Intent is for given site to update / augment `events` as desired. It is called at the end of `get_events()`.

### CODE REVIEW

**Comment by dphoria** [comment]
In #35 , I thought I would move adding `Person.seat.roles` from `inject_known_data()` to `post_process_ingestion_models()`.

```python
# now that we have seat from static hard-coded data
# we can bring in seat.roles (OfficeRecords from Legistar API)
if person.seat and not person.seat.roles:
    person.seat.roles = self.get_roles(
        get_legistar_person(
            client=self.client_name,
            person_id=person.external_source_id,
            use_cache=True,
        )[LEGISTAR_PERSON_ROLES]
    )
```

I changed my mind for a few reasons.

1. Want to keep base `post_process_ingestion_models()` empty.
2. Because Legistar API does not provide `Seat` information at all, it is impossible to set `Person.seat.roles` without injecting `Person.seat` using `known_persons`, anyway. i.e. there is no scenario where we _don't_ inject `Seat` into a `Person`, but can add `Seat.roles` to that `Person`.
3. Iterating through all the ingestion models again to get at all the `Person`s to set `Person.seat.roles` in `post_process_ingestion_models()` just seems too unnecessary.

**Comment by dphoria** [comment]
Please ignore the few commits since previous comment. Checked in code to the wrong (this) branch.

**Review by JacksonMaxfield — APPROVED** [review]
This all looks good to me. Will wait for @sarahjliu's review before merge.

**Pull request merged**
